README.md

```diff
@@ -82,7 +84,210 @@ Requirements:
+       CMake, choose [Makefile](#build-with-makefile)
+
+
+Build with CMake:
+-----------------
+CMake is a configure tool which allows you to out-of-source build.
+Aladdin Requires CMake newer than 2.8.12. By default, CMake
+searches for LLVM 3.4.
+
+
+1. Set `$BOOST_ROOT`,  `TRACER_HOME`, `LLVM_HOME` to let Aladdin know where the tools and libraries are.
+
+   ```
+   export BOOST_ROOT=/your/path/to/boost
+   export TRACER_HOME=/your/path/to/LLVM-Tracer
+   export LLVM_HOME=/your/path/to/llvm
+   ```
+
+2. Use CMake to configure Aladdin and build libraries and executables
+
+   ```
+   mkdir /where/you/want/to/put/your/binary
+   cd /where/you/want/to/put/your/binary
+   cmake /your/path/to/Aladdin
+   make -j4
+   ```
+
+3. (Optional) Run tests with `ctest` command
+
+   **CAUTION : Takes a long time**
+
+   **Suggest running part of them, like `"triad"` or `"unit_test"`.**
+
+   There are 1397 tests. On Intel Xeon E5-2650 2.60GHz,
+   execution in 32 jobs takes 39 minutes,
+   and execution in sequential takes 6.5 hours.
+
+
+   Runs all the tests in sequential.
+   ```
+   ctest
+   ```
+
+   Runs all the tests in parallel. You can mix this option with test
+   filtering option to speed up tests.
+   ```
+   ctest -j JOB_NUM
+   ```
+
+   Runs `triad` tests. Use `-R` flag to denotes the regular expression to match.
+   ```
+   ctest -R triad
+   ```
+
+   Runs unit-tests
+   ```
+   ctest -R unit_test
+   ```
+
+   If just want to know what -R `triad` matches. There is no actual execution of tests
+   ```
+   ctest -R triad --show-only
+   ```
+
+4. Available test names of CTest
+
+   CMake collects available test cases and generate a list of configs,
+   then generate bash shell scripts for each tests.
+   `ctest` command for Aladdin executable usually runs these bash scripts.
+
+   Besides tests using Aladdin, CMake also build tests which generates
+   `dynamic_trace.gz`. It is named after just the case name without the
+   config settings like `triad` or `fft`. CMake also applies the python
+   script `plot_space.py` to plot design space. They are named in
+   `X_data_plot` scheme.
+
+   Of course, if multiple of tests are active at the same
+   time, CTest will schedule all Aladdin tests according to predefined
+   dependencies. Plot tests depend on Aladdin tests and Aladdin tests
+   depend on LLVM-Tracer tests.
+
+   ------------------------------------------------
+
+   The available testcases:
+      * bb_gemm, fft, md, pp_scan, reduction, stencil, triad, ss_sort
+
+   The generated design space goes through:
+
+     Config               |   range
+     ---------------------|---------------------------
+     PARTITION_RANGE      |   [1 2 4 8 16 32 64]
+     UNROLL_RANGE         |   [1 2 4 8 16 32 64]
+     PIPELINE_RANGE       |   [1 2]
+     CLOCK_RANGE          |   [1 6]
+
+
+   Partial CTestName-Command table:
+
+     ctest name                    |    command
+     ------------------------------|---------------------------
```

Comment:
Changes to:

   This is a partial CTestName-ShellCommand table.
   The left side is the test names valid in CTest. While the right
   side is the corresponding shell commands. Just a subset of tests.
   Giving hints for how test names are comprised in Aladdin.

     ctest name                    |    shell command
     ------------------------------|---------------------------
     triad                         |    triad-instrumented
     triad_p2_u16_P2_1ns           |    triad.sh 2 16 2 1
     triad_p2_u2_P1_6ns            |    triad.sh 2 2 1 6